File: examples/haskell/greeter.hs
#!/usr/bin/env runhaskell

import Control.Monad (unless)
import System.IO (hFlush, stdout, stdin, hIsEOF)

-- | For each line FOO received on STDIN, respond with "Hello FOO!".
main :: IO ()
main = do
  eof <- hIsEOF stdin
  unless eof $ do
    line <- getLine
    putStrLn $ "Hello " ++ line ++ "!"
    hFlush stdout
    main

File: examples/swift/greeter.swift
#!/usr/bin/env xcrun -sdk macosx swift

import Foundation

while(true){
  var stdin = NSFileHandle.fileHandleWithStandardInput().availableData
  var line  = NSString(data: stdin, encoding: NSUTF8StringEncoding)!
  var name  = line.stringByTrimmingCharactersInSet(NSCharacterSet.newlineCharacterSet())
  print("Hello \(name)!")
  fflush(__stdoutp)
}
